File: Hackerrank/30-Days/day-06/solution.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        for (int i=0;i<n;i++){
            String s = scanner.nextLine();
            //System.out.println(s);
            ProcessString(s);
        }
        scanner.close();
    }
    
    public static void ProcessString(String str){
        char[] ch = str.toCharArray();
        List<Character> odd_char = new ArrayList<Character>();
        List<Character> even_char = new ArrayList<Character>();
        for(int i=0;i<ch.length;i++){
            if (i%2==0){
                even_char.add(ch[i]);
            } else {
                odd_char.add(ch[i]);
            }
        }
        String even_str = Join(even_char);
        String odd_str = Join(odd_char);
        
        String result = String.format("%s %s",even_str,odd_str);  
        System.out.println(result);
    }
    
    public static String Join(List<Character> char_list){
        StringBuilder sb = new StringBuilder();
        for (char ch : char_list) {
            sb.append(ch);
        }
        return sb.toString();
    }
}

File: Hackerrank/30-Days/day-06/solution.js
const util = require('util'); 

function processText(text){
    var ch = text.split('');
    var even_ch = [];
    var odd_ch = [];
    for(var i=0;i<ch.length;i++){
        if (i%2==0){
            even_ch.push(ch[i]);
        } else {
            odd_ch.push(ch[i]);
        }
    }
    var result = util.format('%s %s', even_ch.join(""), odd_ch.join("")); 
    
    return result;
}

function processData(input) {
    //Enter your code here
    var data = input.split('\n');
    //console.log(data);
    for (var i=1;i<data.length;i++){
        console.log(processText(data[i]));
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});
